### ISSUE TITLE
[BUG]: Pluwise plugin crashes after a couple of hours

### ISSUE THREAD

**Issue by kvanhoorn** [plugwise/plugwise-beta#268]
### Describe the bug.

Plugin (homeassistant repo and this repo) loads without any issues but crashes after some hours, I saw some erros about an IPv6 address earlier, but after switching to this repo that error is gone, it just shuts down.

### What version of Home Assistant Core has the issue?

2022.3.8

### What was the last working version of Home Assistant Core?

2021.12.10

### What type of installation are you running?

Home Assistant Core

### How did you install plugwise-beta?

Cloned from GitHub

### What kind of Plugwise device is having issues?

Smile: Anna

### What firmware version is your Plugwise product at?

4.2.1

### Logging

pi@raspberrypi:~ $ grep plugwise /var/log/homeassistant/home-assistant.log
2022-04-05 15:43:05 INFO (SyncWorker_6) [homeassistant.loader] Loaded plugwise from custom_components.plugwise
2022-04-05 15:43:05 WARNING (SyncWorker_6) [homeassistant.loader] We found a custom integration plugwise which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-04-05 15:43:06 INFO (MainThread) [homeassistant.bootstrap] Domains to be set up: {'auth', 'sensor', 'camera', 'light', 'recorder', 'script', 'shopping_list', 'sonarr', 'version', 'tag', 'person', 'api', 'xiaomi_miio', 'hue', 'pi_hole', 'unifi', 'python_script', 'dsmr', 'picnic', 'media_player', 'synology_dsm', 'webhook', 'plex', 'cover', 'rpi_power', 'map', 'device_tracker', 'updater', 'analytics', 'remote', 'input_datetime', 'zone', 'nest', 'brother', 'mind', 'homekit', 'mobile_app', 'met', 'group', 'zeroconf', 'system_health', 'rfxtrx', 'logger', 'tesla_custom', 'history', 'input_select', 'switch', 'device_automation', 'input_boolean', 'sonos', 'binary_sensor', 'mqtt', 'telegram_bot', 'climate', 'shell_command', 'cloud', 'plaato', 'ffmpeg', 'onboarding', 'tts', 'conversation', 'image', 'ssdp', 'shelly', 'ipp', 'notify', 'weather', 'lovelace', 'sun', 'config', 'http', 'input_number', 'cast', 'system_log', 'local_ip', 'ring', 'utility_meter', 'network', 'search', 'logbook', 'rest_command', 'plugwise', 'blueprint', 'automation', 'harmony', 'melcloud', 'websocket_api', 'trace', 'apple_tv', 'diagnostics', 'frontend'}
2022-04-05 15:43:12 INFO (MainThread) [homeassistant.bootstrap] Setting up stage 2: {'homekit', 'mobile_app', 'sensor', 'camera', 'met', 'group', 'light', 'system_health', 'rfxtrx', 'tesla_custom', 'script', 'history', 'shopping_list', 'sonarr', 'input_select', 'version', 'switch', 'input_boolean', 'tag', 'sonos', 'binary_sensor', 'mqtt', 'telegram_bot', 'xiaomi_miio', 'hue', 'climate', 'shell_command', 'pi_hole', 'plaato', 'ffmpeg', 'tts', 'conversation', 'shelly', 'ipp', 'unifi', 'python_script', 'dsmr', 'notify', 'picnic', 'weather', 'media_player', 'sun', 'synology_dsm', 'input_number', 'plex', 'cast', 'brother', 'cover', 'rpi_power', 'local_ip', 'map', 'ring', 'device_tracker', 'utility_meter', 'logbook', 'updater', 'rest_command', 'remote', 'input_datetime', 'zone', 'nest', 'plugwise', 'blueprint', 'automation', 'harmony', 'melcloud', 'trace', 'apple_tv', 'mind'}
2022-04-05 15:43:30 INFO (MainThread) [homeassistant.setup] Setting up plugwise
2022-04-05 15:43:30 INFO (MainThread) [homeassistant.setup] Setup of domain plugwise took 0.0 seconds
2022-04-05 15:43:48 INFO (MainThread) [homeassistant.components.binary_sensor] Setting up binary_sensor.plugwise
2022-04-05 15:43:48 INFO (MainThread) [homeassistant.components.climate] Setting up climate.plugwise
2022-04-05 15:43:48 INFO (MainThread) [homeassistant.components.number] Setting up number.plugwise
2022-04-05 15:43:48 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.plugwise
2022-04-05 15:43:48 INFO (MainThread) [homeassistant.components.switch] Setting up switch.plugwise
2022-04-05 15:43:48 INFO (MainThread) [homeassistant.components.select] Setting up select.plugwise
2022-04-05 20:36:33 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Anna' for plugwise integration not ready yet: Failed connecting to the Plugwise Smile; Retrying in background

**Comment by bouwew** [comment]
@kvanhoorn there is no reliable connection between the Smile and your HA server: "failed connecting".

**Comment by kvanhoorn** [comment]
Well, an hour later it was restored by itself, but I have never seen this happen so often.
Both devices are connected via cable..

**Comment by bouwew** [comment]
You could show the debug-logging for Plugwise-beta, add this to configuration.yaml:
```
logger:
  default: warn
  logs:
    custom_components.plugwise: debug
    plugwise.smile: debug
```

Maybe that will provide more info.

Also, the connection = cable might be good but the webserver on the Smile can also be the root-cause of the issue.
When looking at where the `failed connecting` message comes from in the code, it's connected to a problem with the server.

**Comment by bouwew** [comment]
Maybe your problem is the same: https://github.com/home-assistant/core/issues/69896

**Comment by kvanhoorn** [comment]
I didn't have the issue anymore since 8 days ago.

**Comment by bouwew** [comment]
Please test plugwise-beta v0.22.3a3 or newer.

**Comment by bouwew** [comment]
@kvanhoorn Is this issue now really fixed for you in plugwise-beta v0.22.3 or newer? Please confirm.

**Comment by bouwew** [comment]
Closing this as no response from the issuer.
